src/Navigation.fs

```diff
@@ -0,0 +1,82 @@
+module Polaris.Navigation
+
+open Fable.Core
+open Fable.Core.JsInterop
+open Fable.React
+open Polaris.Shared
+
+type ItemType = {
+    icon: BundledIcon option
+    label: string
+    url: string
+}
+
+type SectionType = 
+    abstract fill : bool with set
+    abstract items : ResizeArray<ItemType> with set
+    abstract title : string with set
+
+type SubNavigationSectionItem = {
+    label: string
+    url: string
+    disabled: bool
+}
+
+type NavigationSectionItem =
+    abstract label : string with set
+    abstract url : string with set
+    abstract matches : bool with set
+    abstract exactMatch : bool with set
+    abstract matchPaths : ResizeArray<string> with set
+    abstract excludePaths : ResizeArray<string> with set
+    abstract icon : BundledIcon with set
+    abstract badge : string with set
+    abstract disabled : bool with set
+    abstract ``new``: bool with set
+    abstract accessibilityLabel : string with set
+    abstract selected : bool with set
+    abstract subNavigationItems : ResizeArray<SubNavigationSectionItem> with set
+    abstract onClick : (unit -> unit) with set
+
+type NavigationSectionAction =
+    abstract icon : BundledIcon with set
+    abstract accessibilityLabel : string with set
+    abstract onClick : (unit -> unit) with set
+
+type NavigationSectionRollup = {
+    after: int
+    view: string
+    hide: string
+    activePath: string
+}
+
+type RequiredNavigationProps = {
+    Location: string
+}
+
+type NavigationProps =

```

Comment:
I see you dindt add `children`, do you intentionally never plan to add those ? 
They are almost everywhere in the shopify library

Reply:
This one is missing `userMenu`